spec/models/grda_warehouse/tasks/identify_duplicates_spec.rb

```diff
@@ -7,123 +7,196 @@
   let!(:client_in_source) { create :grda_warehouse_hud_client, data_source: source_data_source }
   let!(:client_in_destination) { create :grda_warehouse_hud_client, data_source: destination_data_source }
 
-  it 'recognizes an obvious match' do
-    expect(check_for_obvious_match(client_in_source)).to be client_in_destination.id
-  end
-
-  describe 'obvious match processing with a split' do
-    let!(:split) { create :grda_warehouse_client_split_history, split_from: client_in_source.id, split_into: client_in_destination.id }
-
-    it 'does not return an obvious match if it was split' do
-      expect(check_for_obvious_match(client_in_source)).to be_nil
+  describe 'When matching is enabled' do
+    before(:all) do
```

Comment:
Can this be in a before each rather than all? That way db changes will occur within a transaction and not impact other tests. Also probably cleanup cache on the way out?

```suggestion
    after(:each) { @config.invalidate_cache }

    before(:each) do    
```